File: apps/mobile/fix.py
import os, re

for dp, dn, fn in os.walk('apps/mobile/lib'):
    for f in fn:
        if f.endswith('.dart'):
            path = os.path.join(dp, f)
            with open(path, 'r', encoding='utf-8') as file:
                content = file.read()
            
            # Replace ApiService.xxx('/api/... with ApiService.xxx('/...
            new_content = re.sub(r"ApiService\.(get|post|put|delete|getList)\('/api/", r"ApiService.\g<1>('/", content)
            
            if new_content != content:
                with open(path, 'w', encoding='utf-8') as file:
                    file.write(new_content)
                print(f"Fixed {path}")

File: apps/mobile/fix.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else {
            if (file.endsWith('.dart')) results.push(file);
        }
    });
    return results;
}

const files = walk('lib');
files.forEach(file => {
    const content = fs.readFileSync(file, 'utf8');
    const newContent = content.replace(/ApiService\.(get|post|put|delete|getList)\('\/api\//g, "ApiService.$1('/");
    if (content !== newContent) {
        fs.writeFileSync(file, newContent, 'utf8');
        console.log(`Fixed ${file}`);
    }
});
